app/src/main/java/com/neo/speaktouch/utils/Reader.kt

```diff
@@ -22,159 +22,74 @@ import android.content.Context
 import androidx.core.view.accessibility.AccessibilityNodeInfoCompat
 import com.neo.speaktouch.R
 import com.neo.speaktouch.model.Type
-import com.neo.speaktouch.utils.extension.filterNotNullOrEmpty
-import com.neo.speaktouch.utils.extension.getLog
 import com.neo.speaktouch.utils.extension.ifEmptyOrNull
 import com.neo.speaktouch.utils.extension.iterator
-import timber.log.Timber
+import com.neo.speaktouch.utils.extension.toText
 import javax.inject.Inject
 
 class Reader @Inject constructor(
     private val context: Context
 ) {
-    fun getContent(
-        nodeInfo: AccessibilityNodeInfoCompat
-    ) = getContent(
-        nodeInfo,
-        Level.Text(
-            mustReadSelection = true,
-            mustReadType = true
-        )
-    )
 
-    private fun getContent(
+    fun readContent(
         node: AccessibilityNodeInfoCompat,
-        level: Level
-    ): String {
-        Timber.d(
-            node.getLog(
-                "level: $level"
-            )
-        )
+        options: Options = Options()
+    ) = with(node) {
 
-        return when (level) {
-            is Level.Text -> with(node) {
-                val content = contentDescription.ifEmptyOrNull {
-                    text.ifEmptyOrNull {
-                        hintText.ifEmptyOrNull {
-                            getContent(
-                                node,
-                                Level.Children
-                            )
-                        }
-                    }
+        val content = contentDescription.ifEmptyOrNull {
```

Comment:
**to do:** Behavior in other readers.

Reply:
Well, just like talkback, speak touch will ignore the contentDescription for an edit field. And for now, the contentDescription will prevail in other situations, I will open a discussion about this.

Reply:
1. Talkback always speaks the `hintText`, regardless of whether there is text or not. However, I think it would be interesting to announce the hint only when the edit box is empty, just as it is visually.
2. Talkback ignores the `contentDescription` for the edit field. This makes sense, since the hint is the appropriate choice in this case [^1].

[^1]: https://developer.android.com/guide/topics/ui/accessibility/principles#editable

Reply:
There is a lack of support for content pairs in Speak Touch: https://developer.android.com/guide/topics/ui/accessibility/principles#content-pairs

Reply:
> There is a lack of support for content pairs in Speak Touch: https://developer.android.com/guide/topics/ui/accessibility/principles#content-pairs

I noticed that Jieshou and Prudence also do not have it.

Reply:
> There is a lack of support for content pairs in Speak Touch: https://developer.android.com/guide/topics/ui/accessibility/principles#content-pairs

I created an issue for this #118

Reply:
Changes made in commit 1f513528